tests_onehotencoder.py

```diff
@@ -0,0 +1,236 @@
+# -*- coding: utf-8 -*-
+
+"""
+Copyright (c) 2014, Matthias Feurer
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions are met:
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of the <organization> nor the
+      names of its contributors may be used to endorse or promote products
+      derived from this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL <COPYRIGHT HOLDER> BE LIABLE FOR ANY
+DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+"""
+
+import unittest
+
+import numpy as np
+import scipy.sparse
+from sklearn.utils.testing import assert_array_almost_equal
+
+from tpot.operators.preprocessors._auto_sklearn_onehotencoder import _OneHotEncoder as OneHotEncoder
```

Comment:
You can run the tests by running the nosetests command in the base TPOT
repo directory. That will pick up on all the unit tests.

On Friday, September 2, 2016, Scott Riccardelli <notifications@github.com>
wrote:

> In tests_onehotencoder.py
> <https://github.com/rhiever/tpot/pull/256#discussion_r77372535>:
>
> > +DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
> > +(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
> > +LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
> > +ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
> > +(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
> > +SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
> > +
> > +"""
> > +
> > +import unittest
> > +
> > +import numpy as np
> > +import scipy.sparse
> > +from sklearn.utils.testing import assert_array_almost_equal
> > +
> > +from tpot.operators.preprocessors._auto_sklearn_onehotencoder import _OneHotEncoder as OneHotEncoder
>
> I was executing the test file directly: python tests_onehotencoder.py
>
> How do you typically run tests.py?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/rhiever/tpot/pull/256/files/3dc843edcc320ea96e470943502a49a348428442#r77372535>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABo7t6J0ZQIxPGpnqbfEbv8Iwth1Q-QDks5qmEtlgaJpZM4JzTDj>
> .
>


-- 
Cheers,

Randal S. Olson, Ph.D.
Senior Data Scientist, Institute for Biomedical Informatics
University of Pennsylvania

E-mail: rso@randalolson.com | Twitter: @randal_olson
<https://twitter.com/randal_olson>
http://www.randalolson.com